network/network_test.go

```diff
@@ -742,91 +739,122 @@ func TestNetwork_Reprocess(t *testing.T) {
 		return ctx, eventManager
 	}
 
-	subscribe := func(counter *int, eventManager events.Event) {
-		conn, _, _ := eventManager.Pool().Acquire(context.Background())
-		stream := events.NewDisposableStream("REPROCESS_test", []string{"REPROCESS.*"}, 10)
-		_ = stream.Subscribe(conn, "test", "REPROCESS.*", func(msg *nats.Msg) {
-			foundMutex.Lock()
-			defer foundMutex.Unlock()
-			*counter++
-			err := msg.Ack()
-			if !assert.NoError(t, err) {
-				t.Fatal(err)
+	subscribe := func(ctx context.Context, t *testing.T, eventManager events.Event) chan *nats.Msg {
+		messages := make(chan *nats.Msg, 99)
+
+		conn, _, err := eventManager.Pool().Acquire(ctx)
+		if err != nil {
+			t.Fatal("queue client unavailable:", err)
+		}
+		err = events.NewDisposableStream("REPROCESS_test", []string{"REPROCESS.*"}, 10).Subscribe(conn, "test", "REPROCESS.*", func(m *nats.Msg) {
+			select {
+			case messages <- m:
+				break // collected
+			default:
+				t.Error("subscription reception exceeds buffer capacity", cap(messages))
+			}
+
+			err := m.Ack()
+			if err != nil {
+				t.Error("subscription reception:", err)
 			}
 		})
+		if err != nil {
+			t.Fatal("test subscription error:", err)
+		}
+
+		return messages
 	}
 
-	t.Run("ok", func(t *testing.T) {
-		ctx, eventManager := setup(t)
-		ctx.state.EXPECT().FindBetweenLC(gomock.Any(), uint32(0), uint32(1000)).Return([]dag.Transaction{tx}, nil)
-		ctx.state.EXPECT().ReadPayload(gomock.Any(), tx.PayloadHash()).Return([]byte("payload"), nil)
+	t.Run("hits", func(t *testing.T) {
+		ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+		defer cancel()
 
-		var found int
-		subscribe(&found, eventManager)
+		setup, eventManager := newSetup(t)
+		setup.state.EXPECT().FindBetweenLC(gomock.Any(), uint32(0), uint32(1000)).Return([]dag.Transaction{tx}, nil)
+		setup.state.EXPECT().ReadPayload(gomock.Any(), tx.PayloadHash()).Return([]byte("payload"), nil)
 
-		ctx.network.Reprocess("application/did+json")
+		messages := subscribe(ctx, t, eventManager)
 
-		test.WaitFor(t, func() (bool, error) {
-			foundMutex.Lock()
-			defer foundMutex.Unlock()
-			return found > 0, nil
-		}, 100*time.Millisecond, "timeout waiting for event")
+		_, err := setup.network.ReprocessContentType(ctx, "application/did+json")
+		if err != nil {
+			t.Error("reprocess error:", err)
+		}
+
+		if len(messages) == 0 {
+			t.Error("subscription received no messages")
+		}
 	})
 
-	t.Run("ignores other transactions", func(t *testing.T) {
-		ctx, eventManager := setup(t)
-		ctx.state.EXPECT().FindBetweenLC(gomock.Any(), uint32(0), uint32(1000)).Return([]dag.Transaction{tx}, nil)
-		ctx.state.EXPECT().ReadPayload(gomock.Any(), tx.PayloadHash()).Return([]byte("payload"), nil)
+	t.Run("mismatch", func(t *testing.T) {
```

Comment:
not really a descriptive test name

Reply:
The test name is `ReprocessWithContext/mismatch`. What would be a better description?